python-package/lightgbm/basic.py

```diff
@@ -1476,6 +1483,7 @@ def __init__(
         reference: Optional["Dataset"] = None,
         weight: Optional[_LGBM_WeightType] = None,
         group: Optional[_LGBM_GroupType] = None,
+        position: Optional[_LGBM_PositionType] = None,
```

Comment:
Please move this argument to the end of the signature, after `free_raw_data`. Adding a new argument in the middle of the signature is a breaking change from the perspective of any code that is using only positional arguments, like

```python
dtrain = lgb.Dataset(data, label, reference, weight, group, init_score)
```

Reply:
Done via 0db4caf